<a href="https://colab.research.google.com/github/shivakumardodda/shivakumardodda/blob/data-science/Hypothesis_testing_assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**DODDA SHIVA KUMAR**

**Hypothesis Testing Process**

**1.Hypothesis testing:**Hypothesis testing is the statistical method which is used by analyst to test on assumption regarding the population data by using sample data.

H0=Null hypothesis.
H1=Alternative hypothesis.

Sample data:The data which is subset of the population data and its a peice of the data which is cost effective and time saving process

**2.Caluclate test statistic:**

Assumption:

H0=Cost model not accurate

H1=Cost model is accurate

find the which is correct.

if t-statistical value>t-critical value (H1 is true)

if t-statistical value<t-critical value(H0 is true)

In [ ]:
# Importing Libraries
from scipy import stats
from scipy.stats import norm
import math


In [ ]:
#Data Provided:
sample_mean = 3050  # in Rs.
theoretical_mean = 4000  # in Rs.
std_dev = 5 * 25  # in Rs.
sample_size = 25
alpha = 0.05

In [ ]:
#2 Calculate the test statistic
standard_error = std_dev / math.sqrt(sample_size)
test_statistic = (sample_mean - theoretical_mean) / standard_error
test_statistic

-38.0

In [ ]:
# Degrees of freedom
dof = sample_size - 1
dof

24

**3.Determine the critical value:**

alpha value=0.05
degress of freedom(df)=n-1=24

In [ ]:
from scipy import stats
t_critical=stats.t.ppf(1-0.05,24)
t_critical

1.7108820799094275

**4.Make a Decision:**

By comparing the t_stastical value and t_critical value,we get to know about the model is working accurate or not.whether your null hypothesis is to be rejected or not

In above caluculation we found,

"t_statistical value =-38"

"t_critical value =1.71088"

t_statistical < t_critical value


In [ ]:
#4 Decision
if test_statistic > t_critical:

    print("Reject the null hypothesis")
else:
    print("Fail to reject the null hypothesis")

Fail to reject the null hypothesis


**5.Conclusion:**

From above step(4) we conclued that t_statistical < t_critical,

So from this we got to know about the "Null hypothesis is "True" i.e. the avidance is supporting that weekly cost model is accurate.

Therfore (H0) Weekly cost model is working accuratly.